src/argilla/server/apis/v0/handlers/users.py

```diff
@@ -54,3 +57,20 @@ def create_user(
     user = accounts.create_user(db, user_create)
 
     return User.from_orm(user)
+
+
+@router.delete("/users/{user_id}", response_model=User, response_model_exclude_none=True)
+def delete_user(
+    *, db: Session = Depends(get_db), user_id: UUID, current_user: User = Security(auth.get_user, scopes=[])
+):
+    user = accounts.get_user_by_id(db, user_id)
+    if not user:
+        # TODO: Forcing here user_id to be an string.
+        # Not casting it is causing a `Object of type UUID is not JSON serializable`.
+        # Possible solution redefining JSONEncoder.default here:
+        # https://github.com/jazzband/django-push-notifications/issues/586
+        raise EntityNotFoundError(name=str(user_id), type=User)
```

Comment:
I think adding `json_encoders`  to out error class models should work.

Take a look here: https://docs.pydantic.dev/usage/exporting_models/#json_encoders

Reply:
I will take a look.

Reply:
We will resolve this once we review the error classes